NOnion/Network/TorGuard.fs

```diff
@@ -24,9 +24,16 @@ type TorGuard private (client: TcpClient, sslStream: SslStream) =
         async {
             let tcpClient = new TcpClient ()
 
-            do!
-                tcpClient.ConnectAsync (ipEndpoint.Address, ipEndpoint.Port)
-                |> Async.AwaitTask
+            try
+                do!
+                    tcpClient.ConnectAsync (ipEndpoint.Address, ipEndpoint.Port)
+                    |> Async.AwaitTask
+            with
+                | :? System.AggregateException as aggException ->
+                    match aggException.InnerException with
+                    | :? SocketException as ex ->
+                        return raise (GuardConnectionFailedException ex)
+                    | _ -> return raise aggException
```

Comment:
you should do a reraise here (and as you're in an async job you need to bring the workaround that we have for this in geewallet, in FSharpUtil)